### DISCUSSION TITLE
Is it possible to run this without the rest of automatic1111?

### DISCUSSION THREAD

**Discussion by SebSaga** [thygate/stable-diffusion-webui-depthmap-script#70] [Q&A]
Auto1111 keeps updating with bugs, I was hoping there would be a way to just run this extension so I can batch processes depth maps from images.

Any help would be appreciated 🙏

**Comment by thygate** [comment]
You can run the individual processes locally from their cloned repos or in colabs .. 

I was contemplating maybe making it stand alone in the future, but there are no plans to do so currently.

What does your workflow look like ? only depth ? boost ? stereo ? ...

Usually when there is a new bug in the webui it is fixed within a day or so ..

**Reply by SebSaga** [reply]
Sorry, I'm a total noob and don't know what you mean by "You can run the individual processes locally from their cloned repos" Where/what is the cloned repo? I might consider learning how to run it in a colab book. It's all so new to me but I'm very willing to learn :)

My current workflow is using the depth tab's batch processing to point to an input and output directory, and yes I only use it to generate depth images from input images which I already have. I use res101 I think it's called? I would check but my Auto1111 isnt working LOL and yes, I keep boost turned on because I think that makes better results? As you might be able to tell, I have no idea what I'm doing haha!

I gotta say thank you though. You've been super fast to reply to both my posts! 

By the way, if you do end up making it a stand alone app, I would definitely be willing to pay for it. If it's priced right of course!

**Reply by thygate** [reply]
The goal is to make code from several papers in (code-) repo(sitories) with interesting features (depth maps from a single image, stereo images, 3d reconstruction, ..) available to as many people as possible with an easy install and workflow. The A1111 Webui provides a lot under the hood. Going stand-along would be quite the undertaking for a single dev, so it's unlikely to happen.

For your workflow the current options are this extension, or :

You can run BoostingMonocularDepth with res101 (leres) or midas models up to 3.0 on their colab on a remote machine in the google cloud, linked here : https://colab.research.google.com/github/compphoto/BoostingMonocularDepth/blob/main/Boostmonoculardepth.ipynb

or you can clone the repository locally, and run on your machine. You can clone the repo with `git clone https://github.com/compphoto/BoostingMonocularDepth.git` on the command line in the folder you want to clone into (download a copy of said repo).
Install all the dependencies as explained in their [readme](https://github.com/compphoto/BoostingMonocularDepth#setup), and run the process with leres like this `python run.py --Final --data_dir PATH_TO_INPUT --output_dir PATH_TO_RESULT --depthNet 2`

As you can tell there are a lot of dependencies and collisions between certain packages with certain versions. That's why integration into the A1111 webui makes a lot of sense, as the install is easy and straightforward.

If you want to learn and get stuck, you're welcome to ask questions.

**Reply by wes-kay** [reply]
@thygate 👍 

I looked through the Boost run.py to see if there's any flags to flip the image from --depthNet 2 from black + z to white - z but there is none, in your Automatic1111 implementation do you just flip the black and white for your leres process?

**Reply by thygate** [reply]
@wes-kay Yes.

**Reply by tufeixp** [reply]
Can it work with ComfyUI?
